src/plugins/discover/public/application/main/components/top_nav/get_top_nav_links.tsx

```diff
@@ -75,16 +79,15 @@ export const getTopNavLinks = ({
       defaultMessage: 'Alerts',
     }),
     run: async (anchorElement: HTMLElement) => {
-      const updatedDataView = await persistDataView(dataView);
-      if (updatedDataView) {
-        openAlertsPopover({
-          I18nContext: services.core.i18n.Context,
-          anchorElement,
-          searchSource: savedSearch.searchSource,
-          services,
-          savedQueryId: state.appStateContainer.getState().savedQuery,
-        });
-      }
+      openAlertsPopover({
```

Comment:
"Manage rules and connectors" link in the popover reloads the page when pressed and user loses their ad hoc data views. Can we change it to a navigation without a page reload?

<img width="539" alt="Screenshot 2022-10-18 at 13 06 57" src="https://user-images.githubusercontent.com/1415710/196413760-6abbb427-82e3-48df-a81e-05265b264bfe.png">

Reply:
The way it currently works is that ad-hoc ones are scoped to the plugin they're created in, unless we provide special functionality like the Discover->Lens, Lens->Discover. I do agree that some sort of fallback functionality, restoring e.g. the current ad-hoc one when reloading would be neat. I can see plenty of this requests incoming, one users start to work with ad-hoc ones.